src/Libraries/PythonNodeModelsWpf/PythonNode.cs

```diff
@@ -286,7 +314,13 @@ private void PythonEnginesChanged(object sender, NotifyCollectionChangedEventArg
             {
                 foreach (var item in e.NewItems)
                 {
-                    AddPythonEngineToMenuItems((item as PythonEngine).Name);
+                    if(item is PythonEngine newEngine)
+                    {
+                        if (pythonEngineVersionMenu.Items.Cast<MenuItem>().Any(x => x.Header == newEngine.Name)){
```

Comment:
I think you are already checking this inside the AddPythonEngineToMenuItems method